### PR TITLE
feat(monitors): Adds MonitorEnvironment to check_monitor + update endpoint

### LINKED ISSUE #45022: Crons: Update `check_monitor` task to dual write into monitor environments
The job needs to be updated to query on the monitor environments, both for missed checkins and for checkins that have run for too long.

It will need to dual write into both the monitor and monitor environment.

### PR DESCRIPTION
**Pull request by rjo100**
Adds `MonitorEnvironment` to `check_monitor` and check in update endpoint.

Closes https://github.com/getsentry/sentry/issues/45022

### CODE REVIEW

File: src/sentry/monitors/consumers/check_in.py [hunk 1]
```diff
@@ -37,6 +43,15 @@ def process_message(message: Message[KafkaPayload]) -> None:
                 logger.debug("monitor does not exist: %s", params["monitor_id"])
                 return
 
+            environment_name = params.get("environment")
+            if not environment_name:
+                environment_name = "production"
+            environment = Environment.get_or_create(project=monitor.project, name=environment_name)
```
Comment (rjo100) [inline src/sentry/monitors/consumers/check_in.py:49]: this logic needs cleanup/consolidation later

File: src/sentry/monitors/consumers/check_in.py [hunk 2]
```diff
@@ -98,6 +115,9 @@ def process_message(message: Message[KafkaPayload]) -> None:
             Monitor.objects.filter(id=monitor.id).exclude(last_checkin__gt=start_time).update(
                 **monitor_params
             )
+            MonitorEnvironment.objects.filter(id=monitor_environment.id).exclude(
+                last_checkin__gt=start_time
+            ).update(**monitor_params)
```
Comment (rjo100) [inline src/sentry/monitors/consumers/check_in.py:120]: do we have any tests for this file?

File: src/sentry/monitors/consumers/check_in.py [hunk 3]
```diff
@@ -46,7 +61,7 @@ def process_message(message: Message[KafkaPayload]) -> None:
                 check_in = MonitorCheckIn.objects.select_for_update().get(
                     guid=params["check_in_id"],
                     project_id=project_id,
-                    monitor=monitor,
+                    monitor_environment=monitor_environment,
```
Comment (rjo100) [inline src/sentry/monitors/consumers/check_in.py:64]: this will have to be reverted as the data will not have been backfilled yet

**Review by wedamija — APPROVED** [review]

File: src/sentry/monitors/consumers/check_in.py (hunk 2 above)
Comment (rjo100) [reply src/sentry/monitors/consumers/check_in.py:130]: we do not

**Pull request merged**